src/foam/core/Action.js

```diff
@@ -177,10 +190,21 @@ If empty than no permissions are required.`,
     function maybeCall(x, data) {
       var self = this;
       function call() {
-        self.code.call(data, x, self);
+        var running = self.runningMap.get(data);
```

Comment:
Yeah let's put this in a utility function so we can just call getRunning$(data) both here and createIsEnabled$().  Then its readable and we're guaranteed to only have one running slot per action+data combination.